### ISSUE TITLE
v0.5 Ability to create functions in subdirectories from project root disrupted

### ISSUE THREAD

**Issue by erikerikson** [serverless/serverless#796]
previously you could create all your functions from the root directory of a project:
`sls project create -n myName [...]` [enter]
`cd myName` [enter]
`sls function create` [enter]
`> Serverless: Enter a new function name:`
`my/new/func` [enter]

Expected: new function skeleton in the `[projectRoot]/my/new/func` subdirectory.

Observed:
```
error:   Function name must contain only letters, numbers, hyphens, or underscores. It should not be longer than 20 characters.
```

Exception for the path delimiter character in an OS neutral manner?

**Comment by ac360** [comment]
The new syntax is `sls function create api/users/usersShow` without the `-n` flag.  I understand there is some confusion here and some UX quirks.  Thanks as always @erikerikson

**Comment by erikerikson** [comment]
Sorry, I was trying to make the issue clear and ended up miscommunicating.  This error was seen in interactive mode.  Will change original description.

Previously was:
previously you could create all your functions from the root directory of a project:
`sls project create -n myName`
`cd myName`
`sls function create -n my/new/func`

**Comment by erikerikson** [comment]
I can confirm that the `slsDev function create my/new/func` command form accepts the path delimiters.

**Comment by erikerikson** [comment]
the two methods should probably validate using the same code.  :)

**Comment by ac360** [comment]
Thanks @erikerikson :)  Fixed in 01120b8944a61d7f4a48c4dcf0c5627b6dd52f7f
